clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientBuilderImpl.java

```diff
@@ -212,11 +219,26 @@ public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credenti
 
   @Override
   public ZeebeClient build() {
+    applyOverrides();
     applyDefaults();
 
     return new ZeebeClientImpl(this);
   }
 
+  private void applyOverrides() {
+    if (!insecureConnSetAsProperty && Environment.system().isDefined(PLAINTEXT_CONNECTION_VAR)) {
+      if (Environment.system().get(PLAINTEXT_CONNECTION_VAR).equals("true")) {
```

Comment:
Also if we start doing this, let's standardize it and add a `getBool` to the `Environment` class which does this check for us.

Reply:
Do you think that should be in the Environment? For the environment, every variable is a string so it seems strange to be treating it as a bool without any semantic information about the variable (the builder knows what the variable actually is). Or am I misunderstanding?

Reply:
A `Environment#getBool(String)` still lets the caller define the semantics of the variable.